### DISCUSSION TITLE
PDF generator

### DISCUSSION THREAD

**Discussion by OSAwebsite** [code4recovery/12-step-meeting-list#1142] [Q&A] [answered]
I am trying to understand the description of the PDF generator in the plugin.
I failed... 

I am not sure what to do with the URL that I am given.
I have never worked with Google docs.
Is there an 'explain it to me as if I were a 9 year old' version that I can follow step by step in an easier way?

Thanks!

**Comment by kiyote33** [comment]
Hi @OSAwebsite,
It's hard to understand what you are getting at, but I'll make a few assumptions and give it a try! Please let me know if I've missed the boat. My understanding of what you are trying to do is to use the Code For Recovery "PDF Generator Service" to produce a printable copy of your website meeting list? You have followed our documentation on the [Code For Recovery website](https://code4recovery.org/) which lead you to the page at [https://github.com/code4recovery/pdf](https://github.com/code4recovery/pdf.)  On that page you can build a URL for your website (which follows this format: https://YOUR_DOMAIN_NAME//wp-admin/admin-ajax.php?action=meetings), configure the options for your output, and then press the Generate button to produce a PDF file which should show up in your downloads folder as directory.pdf. Is this what you are wanting to do? I hope I've answered your question...

**Comment by OSAwebsite** [comment]
Hi Wayne,
Thanks for taking the trouble to reply to my question.
I think I found the cause of my issue, but I am a little disappointed in
what the list looks like.
I don't think any development was done on it since the last time I tried.
My issue was solved in the Settings, by choosing 'Open' instead of
'Limited' in the 'Feed' section.
I am actually not entirely sure what is meant by that, if you do I'd love
to know.

My main issue with the list is that I am not sure it is set up in the way a
newcomer would look for a meeting.
Maybe it is, maybe it is not.
I remember an earlier conversation about it with me asking to divide it by
district.
The reply at the time was that district mean nothing to the newcomer and
that is probably true.

But as it is sorted now, it seems to be by place of meeting first and then
by timeslot. That makes it a bit hard to consult.
I would much rather see it sorted by day and time of the day and THEN by
place of meeting.
It feels like a bit of a disappointment to be looking for a meeting at a
certain time and then, although finding the place I prefer, finding that
the time has already passed.
Does that make sense?

Anyway... I guess all people want different things.

In an ideal world I would love to offer the visitors of websites the option
to choose from a number of formats.
Maybe be able to print according to their found set.
Or find meetings for a certain place and just print those.
More options at the front end.

I may be the website coordinator for an area AND a program member, but
there are many who have delegated the website to a professional.
And I cannot imagine that it would be handy to send a generic webmaster a
request to have a full list printed.
Having it available at the front end would be so much better.

But: that might be a nice to have and not a must have.
For now, I do not find the PDF useful. Hoping for some new development in
that area some time!

In service,
Nicky D



-
Yours in service,
Nicky D
Acting Ontario South Website Coordinator
al-anon.alateen.on.ca

*This too will pass...*

**Reply by kiyote33** [reply]
Again I can only assume you are referring to the Feed Sharing options of 'Open' and 'Restricted' in the Feed Management section of the 12 Step Meeting List plugin. Restricted means you need to append a unique key to the URL to access the JSON feed of your meeting list. Open means you don't need a key.

We thank you for your feedback on the PDF. It does help us make improvements when making product revisions.

**Reply by joshreisner** [reply]
we havent made any changes yet regarding feed security. if/when we do, the option will simply go away

**Comment by anchovie91471** [comment, marked as answer]
@OSAwebsite , I am one of the maintainers of PDF. I would be happy to take a look at what you are proposing as a potential enhancement. Could you mock up what you suggest and post it in the PDF issues? https://github.com/code4recovery/pdf/issues

**Comment by OSAwebsite** [comment]
Sure. Will do. Appreciate your help!


-
Yours in service,
Nicky D
Acting Ontario South Website Coordinator
al-anon.alateen.on.ca

*This too will pass...*
